### PR TITLE
 DScanner: Enable warnings for unused parameters ...

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
... and fix `access.d`, `astbase`, `todt.d`, `traits.d` and `typesem.d` for starters. This obviously requires quite a blacklist right now but still covers about 2/3 of all source files.

Bonus: This revealed that the documentation of `checkAccess` did not match its implementation.

CC @wilzbach

### CODE REVIEW

File: src/dmd/todt.d [hunk 1]
```diff
@@ -76,9 +76,9 @@ extern (C++) void Initializer_toDt(Initializer init, ref DtBuilder dtb)
         dtb.nzeros(cast(uint)vi.type.size());
     }
 
-    void visitStruct(StructInitializer si)
+    void visitStruct(StructInitializer /* unused */)
```
Comment (rainers) [inline src/dmd/todt.d]: I would not rename the variables, commenting them is enough indication of them being unused.

File: src/dmd/typesem.d [hunk 2]
```diff
@@ -3444,6 +3444,7 @@ Expression dotExp(Type mt, Scope* sc, Expression e, Identifier ident, int flag)
                 tf.purity = PURE.const_;
                 tf.isnothrow = true;
                 tf.isnogc = false;
+                cast(void) tf;
```
Comment (rainers) [inline src/dmd/typesem.d]: This looks wrong, seems a bug in dscanner?

**Review by rainers — COMMENTED** [review]
Some of the changes are good (e.g. adding module declarations), but I'm not so sure about enabling warnings about unused parameters. Removing them also reduces information provided by intellisense or documentation.

File: src/dmd/typesem.d (hunk 2 above)
Comment (MoonlightSentinel) [reply src/dmd/typesem.d]: Yes,  allthough its already fixed in master (we use an older version of dscanner)

File: src/dmd/todt.d (hunk 1 above)
Comment (MoonlightSentinel) [reply src/dmd/todt.d]: Done

**Comment by MoonlightSentinel** [comment]
> Some of the changes are good (e.g. adding module declarations), but I'm not so sure about enabling warnings about unused parameters. Removing them also reduces information provided by intellisense or documentation.

I think checking for unused parameters should be done because it can reveal violations of a functions contract/documentation (see e.g. `checkAccess`). But DScanner should have something like Javas `SuppressWarnings("unused")` to mark parameters that are reserved for overriding methods.

Anyway, i would be fine with splitting this into different PRs for local/parameters. But limiting this check to local variables is currently not possible (because it was introduced after https://github.com/dlang-community/D-Scanner/issues/786 appeared.

**Comment by MoonlightSentinel** [comment]
PR to enable checks for unused variables only #10714

**Comment by MoonlightSentinel** [comment]
Closing until D-Scanner provides some means to allow unused parameters for documentation purposes.

**Pull request closed without merging**